### ISSUE TITLE
[Bug] Crash on world load

### ISSUE THREAD

**Issue by kaleidopope** [sp614x/optifine#4893]
## Description of Issue
Loading any singleplayer world with Optifine leads to either Minecraft getting stuck on 100% or the game crashing entirely with the Minecraft launcher spitting out this message:
```
The game crashed whilst unexpected error
Error: java.lang.IllegalStateException: Lock is no longer valid
Exit Code: -1
```
This doesn't seem to happen when generating a new world. 

## Steps to Reproduce
1. Create new world in either vanilla or Optifine, save, exit, and quit
2. Load world with Optifine profile

## OptiFine Version
OptiFine_1.16.3_HD_U_G3

## Installation Method
Standalone installer

## Log Files/Crash Reports
[crash-2020-10-08_23.17.04-client.txt](https://github.com/sp614x/optifine/files/5352916/crash-2020-10-08_23.17.04-client.txt)

## Prior Testing
Vanilla seems to work fine, both world generation and world loading.
Tried closing and reopening Minecraft, redownloading both Minecraft 1.16.3 and Optifine, and restarting computer. No dice.

**Comment by sp614x** [comment]
Are you using a Vanilla Tweaks datapack?

**Comment by kaleidopope** [comment]
No, the game is pretty much untouched other than Optifine to my knowledge.

**Comment by LuckyOwl1105** [comment]
I've been having this problem for a couple of months now.

**Comment by kaleidopope** [comment]
Still happening in 1.16.4...

`The game crashed whilst unexpected error
Error: java.lang.IllegalStateException: Lock is no longer valid
Exit Code: -1`

[crash-2020-11-11_09.23.56-client.txt](https://github.com/sp614x/optifine/files/5525760/crash-2020-11-11_09.23.56-client.txt)

Again, I simply downloaded 1.16.4, got the standalone OptiFine 1.16.4_HD_U_G5 installer from the website, and opened a singleplayer world in the new profile. Crashed.

Recently, I actually managed to enter a multiplayer server in 1.16.1 with the 1.16.1 version of Optifine. I haven't tested multiplayer with 1.16.3 or 1.16.4 yet.

**Comment by Horos402** [comment]
I have the same problem... on 1.16.1 there are no problems, with further versions sometimes i manage to get into the worlds by deleting session.lock or creating a new world (hoping don't crash) and then copy-paste the world files (except session.lock) from the precendent world. But it works 20% of the times.

**Comment by COSoundwave** [comment]
I have this issue as well. I don't know if this helps, but I've been looking around, and from what I can tell all the crash reports I've found have the same details section: 

> Details:
> 	All players: 1 total; [ClientPlayerEntity['Transfandomman'/159, l='ClientLevel', x=8.50, y=65.00, z=8.50]]
> 	Chunk stats: Client Chunk Cache: 625, 0
> 	Level dimension: minecraft:overworld
> 	Level spawn location: World: (8,64,8), Chunk: (at 8,4,8 in 0,0; contains blocks 0,0,0 to 15,255,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
> 	Level time: 0 game time, 0 day time
> 	Server brand: ~~NULL~~
> 	Server type: Integrated singleplayer server

 
Down the the spawn point and the location of what other people have said is a corrupted block right next to it, the coordinates are the same.

**Comment by TiarSukaCoding** [comment]
still happening for me on Optifine  1.16.5 HD U G7
[crash-2021-05-15_19.09.04-client.txt](https://github.com/sp614x/optifine/files/6486916/crash-2021-05-15_19.09.04-client.txt)

**Comment by Tradesminator** [comment]
Still happens in 1.16.5 HD U G8.

Without forge
[crash-2021-06-23_02.47.21-client.txt](https://github.com/sp614x/optifine/files/6712784/crash-2021-06-23_02.47.21-client.txt)

With forge 36.1.0 and no other mods installed
[crash-2021-06-23_04.02.13-client.txt](https://github.com/sp614x/optifine/files/6712787/crash-2021-06-23_04.02.13-client.txt)

If it helps, I play minecraft on a external storage drive, because that's where I set the AppData folder. Sometimes the crash does not occur the first time I play after installing optifine.

I tested on preview 1.17 HD U G9 pre21 and works fine for now.

**Comment by ShadoUrufu** [comment]
I'm on version 1.18, version HD U H3, it keeps crashing on world load since I updated. Optifabric still loads things, and it used to work until at least pre7
